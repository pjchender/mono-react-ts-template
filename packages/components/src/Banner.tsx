import { FC } from 'react';
import styled from 'styled-components';

const StyledHeader = styled.h1.attrs({
  'data-testid': 'banner',
})<{ isDark: boolean }>`
  font-size: 24px;
  color: ${props => (props.isDark ? 'white' : 'black')};
`;

interface Props {
  text: string;
}
const Banner: FC<Props> = ({ text }: Props) => {
  return (
    <div>
      <StyledHeader isDark>Hello, {text}</StyledHeader>
    </div>
  );
};

export default Banner;
