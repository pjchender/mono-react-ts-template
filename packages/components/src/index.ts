function components(foo: string, bar: string) {
  return foo + bar
}

export default components;
