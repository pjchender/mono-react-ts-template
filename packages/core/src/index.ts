function core(foo: string, bar: string) {
    return foo + bar
  }

  export default core;
