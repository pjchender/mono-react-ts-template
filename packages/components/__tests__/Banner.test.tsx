import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Banner } from '../src';

test('Banner should show correct text', async () => {
  const banner = render(<Banner text="foo" />);
  const component = await banner.findByTestId('banner');
  expect(component.innerHTML).toBe('Hello, foo');
});

test('Banner snapshot show be the same', () => {
  const tree = renderer.create(<Banner text="foo" />).toJSON();
  expect(tree).toMatchSnapshot();
});
