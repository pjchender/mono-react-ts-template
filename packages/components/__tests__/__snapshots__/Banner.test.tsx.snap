// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner snapshot show be the same 1`] = `
<div>
  <h1
    className="Banner__StyledHeader-sc-13xmge6-0 kiILqj"
    data-testid="banner"
  >
    Hello, 
    foo
  </h1>
</div>
`;
